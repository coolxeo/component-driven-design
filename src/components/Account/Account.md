Account example:

```js
<Account image="/water-cure.jpg" rate="4.8" title="The Water Cure" subtitle="Joanne Ramos" buttonText="Biography" />
```

Account example without rate & button:

```js
<Account image="/water-cure.jpg" title="The Water Cure" subtitle="Joanne Ramos" />
```