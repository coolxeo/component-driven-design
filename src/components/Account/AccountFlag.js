import React from 'react'
import styled from 'styled-components';

const ImageContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 10px;
`;

const Image = styled.div`
  background-image: url(${props => props.image});
  background-size: cover;
  width: 60px;
  height: 60px;
  border-radius: 18px;
`;

const SmallImage = styled(Image)`
  width: 30px;
  height: 30px;
`

function AccountFlag({ image, small }) {
  return (
    <ImageContainer>
      {!small && <Image image={image} />}
      {small && <SmallImage image={image} />}
    </ImageContainer>
  )
}

export default AccountFlag
