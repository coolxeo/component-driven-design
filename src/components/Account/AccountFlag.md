AccountFlag example:

```js
<AccountFlag image="/gbp.png" />
```

AccountFlag Small example:

```js
<AccountFlag image="/gbp.png" small={true} />
```