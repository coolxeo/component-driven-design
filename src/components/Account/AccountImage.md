AccountImage example:

```js
<AccountImage image="/water-cure.jpg" />
```

AccountImage Small example:

```js
<AccountImage image="/water-cure.jpg" small={true} />
```