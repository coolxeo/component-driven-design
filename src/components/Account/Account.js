import React from 'react';
import styled from 'styled-components';

import AccountImage from './AccountImage';
import AccountDetails from './AccountDetails/AccountDetails';

const AccountContainer = styled.div`
  display: flex;
  flex-direction: row;
`

function Account({ image, rate, title, subtitle, buttonText }) {
  return (
    <AccountContainer>
      <AccountImage image={image} />
      <AccountDetails rate={rate} title={title} subtitle={subtitle} buttonText={buttonText} />
    </AccountContainer>
  )
}

export default Account;
