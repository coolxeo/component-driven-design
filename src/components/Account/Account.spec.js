import React from 'react'
import renderer from 'react-test-renderer'

import Account from './Account'

it('Account renders correctly', () => {
  const tree = renderer
    .create(
      <Account image="/water-cure.jpg" rate="4.8" title="The Water Cure" subtitle="Joanne Ramos" buttonText="Biography" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('DoesnÂ´t show the button or rate', () => {
  const tree = renderer
    .create(
      <Account image="/water-cure.jpg" title="The Water Cure" subtitle="Joanne Ramos"/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});