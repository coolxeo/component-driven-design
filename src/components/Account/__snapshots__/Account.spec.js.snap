// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account renders correctly 1`] = `
<div
  className="sc-htoDjs lkdQiJ"
>
  <div>
    <div
      className="sc-bdVaJa fmTyxS"
    />
  </div>
  <div
    className="sc-gzVnrw fSNPds"
  >
    <div
      className="sc-htpNat kMQpYq"
    >
      4.8
    </div>
    <div
      className="sc-bxivhb fxQdBO"
    >
      The Water Cure
    </div>
    <div
      className="sc-ifAKCX zuhiC"
    >
      Joanne Ramos
    </div>
    <div>
      <div
        className="sc-EHOje fEesZF"
      >
        Biography
      </div>
    </div>
  </div>
</div>
`;

exports[`DoesnÂ´t show the button or rate 1`] = `
<div
  className="sc-htoDjs lkdQiJ"
>
  <div>
    <div
      className="sc-bdVaJa fmTyxS"
    />
  </div>
  <div
    className="sc-gzVnrw fSNPds"
  >
    <div
      className="sc-bxivhb fxQdBO"
    >
      The Water Cure
    </div>
    <div
      className="sc-ifAKCX zuhiC"
    >
      Joanne Ramos
    </div>
  </div>
</div>
`;
